server/Mistakes.Journal.Api/Api/Mistakes/Controllers/MistakesController.cs

```diff
@@ -138,20 +197,27 @@ public async Task<ActionResult<ICollection<MistakeWebModel>>> UpdateMistake(Guid
                 .Include(m => m.Tips)
                 .FirstOrDefaultAsync(m => m.Id == mistakeId);
 
-            existingMistake.Name = mistake.Name;
-            existingMistake.Goal = mistake.Goal;
-            existingMistake.Priority = mistake.Priority.GetValueOrDefault();
+            if (existingMistake is null)
+                return NotFound();
+
+            existingMistake.Name = mistake.Name ?? existingMistake.Name;
+            existingMistake.Goal = mistake.Goal ?? existingMistake.Goal;
+            existingMistake.Priority = mistake.Priority ?? existingMistake.Priority;
 
-            var deletedTips = existingMistake.Tips.Where(t => !mistake.Tips.Contains(t.Content));
-            _dataContext.Set<Tip>().RemoveRange(deletedTips);
+            if (mistake.Tips != null)
+            {
+                var deletedTips = existingMistake.Tips.Where(t => !mistake.Tips.Contains(t.Content));
+                _dataContext.Set<Tip>().RemoveRange(deletedTips);
 
-            var newTips = mistake.Tips.Where(t => !existingMistake.Tips.Select(et => et.Content).Contains(t));
-            existingMistake.Tips.AddRange(newTips.Where(t => !string.IsNullOrWhiteSpace(t)).Select(t => new Tip(t)));
+                var newTips = mistake.Tips.Where(t => !existingMistake.Tips.Select(et => et.Content).Contains(t));
+                existingMistake.Tips.AddRange(newTips.Where(t => t.IsPresent())
+                    .Select(t => new Tip(t)));
```

Comment:
What's up with this Select? It looks a bit unreadable. Maybe just add this Where and Select to `newTips` query?